packages/react-querybuilder/src/utils/formatQuery/defaultValueProcessorMongoDBByRule.ts

```diff
@@ -17,80 +17,73 @@ export const defaultValueProcessorMongoDBByRule: ValueProcessorByRule = (
   if (
     operator === '<' ||
     operator === '<=' ||
-    operator === '=' ||
     operator === '!=' ||
     operator === '>' ||
     operator === '>='
   ) {
     const mongoOperator = mongoOperators[operator];
-    return valueIsField
-      ? `{"$expr":{"${mongoOperator}":["$${field}","$${value}"]}}`
-      : `{"${field}":{"${mongoOperator}":${
-          useBareValue ? trimIfString(value) : `"${escapeDoubleQuotes(value)}"`
-        }}}`;
+    return `"${field}":{"${mongoOperator}":${
```

Comment:
Why does this ignore the case where `valueSource === "field"`?

Reply:
Sorry, this valueSource is not clear for me.
`valueSource === "field"` What is the use case for this ? When is the querybuilder generating this?

Reply:
It's for comparing a field to another field, instead of a field to a value. [See documentation here](https://react-querybuilder.js.org/docs/api/valueeditor#value-sources).